### PR TITLE
Match haskell-lsp updates for VFS race condition fixes

### PR DESCRIPTION
**Pull request by alanz**

### CODE REVIEW

File: src/Language/Haskell/LSP/Test/Session.hs [hunk 1]
```diff
@@ -220,8 +221,10 @@ runSessionWithHandles serverIn serverOut serverProc serverHandler config caps ro
   mainThreadId <- myThreadId
 
   let context = SessionContext serverIn absRootDir messageChan reqMap initRsp config caps
-      initState = SessionState (IdInt 0) mempty mempty 0 False Nothing
-      runSession' = runSession context initState
+      initState tmp_dir = SessionState (IdInt 0) (VFS mempty tmp_dir)
+                                       mempty 0 False Nothing
+      runSession' ses = withSystemTempDirectory "lsp-test" $ \tmp_dir ->
```
Comment (mpickering) [inline src/Language/Haskell/LSP/Test/Session.hs]: I added a function to `haskell-lsp` to avoid having to manually call `withSystemTempDirectory`.

Comment (alanz) [reply src/Language/Haskell/LSP/Test/Session.hs]: which is called?

Comment (mpickering) [reply src/Language/Haskell/LSP/Test/Session.hs]: `initVFS` - https://github.com/alanz/haskell-lsp/blob/master/src/Language/Haskell/LSP/VFS.hs#L78

File: src/Language/Haskell/LSP/Test/Session.hs [hunk 2]
```diff
@@ -70,6 +70,7 @@ import System.Directory
 import System.IO
 import System.Process (ProcessHandle())
 import System.Timeout
+import System.IO.Temp
```
Comment (bubba) [inline src/Language/Haskell/LSP/Test/Session.hs]:
```suggestion
```

File: lsp-test.cabal [hunk 3]
```diff
@@ -56,6 +56,7 @@ library
                      , rope-utf16-splay
                      , text
                      , transformers
+                     , temporary
```
Comment (bubba) [inline lsp-test.cabal]:
```suggestion
```

**Pull request merged**